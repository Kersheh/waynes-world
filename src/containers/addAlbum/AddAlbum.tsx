import React, { useState, useEffect } from 'react';
import { useDispatch } from 'react-redux';

import Button from 'components/button/Button';
import IconPlusOutline from 'components/icons/IconPlusOutline';
import { clearSearchQueryAction } from './addAlbumActions';
import Search from './search/Search';
import EditAlbum from './editAlbum/EditAlbum';
import styles from './AddAlbum.module.scss';

const AddAlbumContainer = () => {
  const dispatch = useDispatch();
  const [showSearch, setShowSearch] = useState(false);
  const [showEditAlbum, setShowEditAlbum] = useState(false);

  useEffect(() => {
    dispatch(clearSearchQueryAction());
  }, [dispatch]);

  return (
    <div className={styles.addAlbum}>
      {!showEditAlbum && (
        <>
          <h2>Search to add new album</h2>
          <Search showSearch={showSearch} setShowSearch={setShowSearch} />

          <div className={styles.altText}>
            <h3>or</h3>
          </div>

          <h2>Manually add new album information</h2>
          <div className={styles.createAlbumBtn}>
            <IconPlusOutline />

            <Button
              type="button"
              styleType="basic"
              onClick={() => setShowEditAlbum(true)}
            >
              Create new album
            </Button>
          </div>
        </>
      )}

      {showEditAlbum && (
        <>
          <EditAlbum setShowEditAlbum={setShowEditAlbum} />
        </>
      )}
    </div>
  );
};

export default AddAlbumContainer;
