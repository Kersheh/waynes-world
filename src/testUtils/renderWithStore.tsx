import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose, Reducer } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { render } from '@testing-library/react';

import rootReducer from 'store/reducers';
import rootSaga from 'store/sagas';

const renderWithStore = (component: React.ReactElement) => {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    rootReducer as Reducer,
    compose(applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{component}</Provider>);
};

export default renderWithStore;
