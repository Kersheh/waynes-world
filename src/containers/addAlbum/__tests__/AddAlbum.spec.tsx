import React from 'react';

import renderWithStore from 'testUtils/renderWithStore';
import AddAlbum from '../AddAlbum';

describe('<AddAlbum/> component', () => {
  it('should render', () => {
    const { container } = renderWithStore(
      <AddAlbum />
    );

    expect(container).toMatchSnapshot();
  });
});
