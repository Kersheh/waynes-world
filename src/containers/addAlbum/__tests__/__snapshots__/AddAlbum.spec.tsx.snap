// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddAlbum/> component should render 1`] = `
<div>
  <div
    class="addAlbum"
  >
    <h2>
      Search to add new album
    </h2>
    <div
      class="search"
    >
      <div
        class="searchBtn"
      >
        <svg
          height="24"
          viewBox="0 0 512 512"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M357.079 341.334l94.476 110.73-32.508 27.683-94.222-110.476q-45.714 30.476-100.826 30.476-36.826 0-70.476-14.349t-57.905-38.603-38.603-57.905-14.349-70.476 14.349-70.476 38.603-57.905 57.905-38.603 70.476-14.349 70.476 14.349 57.905 38.603 38.603 57.905 14.349 70.476q0 23.365-5.841 45.714t-16.635 41.651-25.778 35.555zM224 357.079q28.19 0 53.841-11.048t44.19-29.587 29.587-44.19 11.048-53.841-11.048-53.841-29.587-44.191-44.19-29.587-53.841-11.047-53.841 11.047-44.191 29.588-29.587 44.19-11.047 53.841 11.047 53.841 29.588 44.19 44.19 29.587 53.841 11.048z"
            fill="currentColor"
          />
        </svg>
        <button
          class="button basic"
          type="button"
        >
          Artists or albums
        </button>
      </div>
    </div>
    <div
      class="altText"
    >
      <h3>
        or
      </h3>
    </div>
    <h2>
      Manually add new album information
    </h2>
    <div
      class="createAlbumBtn"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24 10 h-10 v-10 h-2 v10 h-10 v2 h10 v10 h2 v-10 h10 z"
          fill="white"
        />
      </svg>
      <button
        class="button basic"
        type="button"
      >
        Create new album
      </button>
    </div>
  </div>
</div>
`;
